src/sys/socket/addr.rs

```diff
@@ -1537,6 +1537,17 @@ impl SockaddrLike for SockaddrStorage {
                 let mut ss: libc::sockaddr_storage = mem::zeroed();
                 let ssp = &mut ss as *mut libc::sockaddr_storage as *mut u8;
                 ptr::copy(addr as *const u8, ssp, len as usize);
+                #[cfg(any(
+                    target_os = "android",
+                    target_os = "fuchsia",
+                    target_os = "illumos",
+                    target_os = "linux"
+                ))]
+                if i32::from(ss.ss_family) == libc::AF_UNIX {
+                    // Safe because we UnixAddr is strictly smaller than
+                    // SockaddrStorage, and we just initialized the structure.
+                    (*(&mut ss as *mut libc::sockaddr_storage as *mut UnixAddr)).sun_len = len as u8;
+                }
```

Comment:
A `libc::sockaddr_storage` was initialized above, but I don't think that means that those bytes represent a valid `UnixAddr`? For example `libc::sockaddr_storage` has [6 padding bytes](https://github.com/rust-lang/libc/issues/3004) on Linux x86-64 that may not be zeroed, and reading those uninitialized bytes as a `UnixAddr` would be UB. I think this would only occur if `len` is less than 8 (`sizeof(u16) + 6`).

Reply:
The docs for `mem::zeroed` say that padding bytes may not be initialized.  But if you look at the source, they are, at least in the current implementation.  So I don't think there's a problem to merge this PR.  But we should still fix the issue you raised with libc.